cdap-hbase-compat-base/src/main/java/co/cask/cdap/replication/TableUpdater.java

```diff
@@ -32,9 +32,16 @@
  * Utility class for Replication Status Tool
  */
 public abstract class TableUpdater {
-
+  private class TimeValue {
+    private long time;
+    private boolean stale;
+    private TimeValue(long time, boolean stale) {
+      this.time = time;
+      this.stale = stale;
+    }
+  }
   private static final Logger LOG = LoggerFactory.getLogger(TableUpdater.class);
-  private HashMap<String, Long> cachedUpdates;
+  private HashMap<String, TimeValue> cachedUpdates;
```

Comment:
Map<String, TimeValue>. Usually use the most generic interface you can rather than a concrete implementation.